spec/std/http/server/handlers/static_file_handler_spec.cr

```diff
@@ -41,13 +48,55 @@ describe HTTP::StaticFileHandler do
     it "should serve file if file mtime is younger" do
       headers = HTTP::Headers.new
       headers["If-Modified-Since"] = HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time - 1.hour)
-      response = handle HTTP::Request.new("GET", "/test.txt")
+      response = handle HTTP::Request.new("GET", "/test.txt", headers)
       response.status_code.should eq(200)
       response.headers["Last-Modified"].should eq(HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time))
       response.body.should eq(File.read("#{__DIR__}/static/test.txt"))
     end
   end
 
+  context "with If-None-Match header" do
+    it "should return 304 Not Modified if header matches etag" do
+      etag = %{W/"#{File.info("#{__DIR__}/static/test.txt").modification_time.epoch.to_s}"}
```

Comment:
You shouldn't generate a etag manually. How it's generated is irrelevant to this spec.

Just send a prior request to get the etag.